src/endpoint/s3/s3_utils.js

```diff
@@ -274,6 +274,40 @@ function set_encryption_response_headers(req, res, encryption = {}) {
     if (kms_key_id) res.setHeader('x-amz-server-side-encryption-aws-kms-key-id', kms_key_id);
 }
 
+function set_range_response_headers(req, res, range, obj_size) {
+    // reply with HTTP 206 Partial Content
+    res.statusCode = 206;
+    const content_range = `bytes ${range.start}-${range.end - 1}/${obj_size}`;
+    dbg.log1('reading object range', req.path, content_range, range);
+    res.setHeader('Content-Range', content_range);
+    res.setHeader('Content-Length', range.end - range.start);
+}
+
+/**
+ * A 'ranged' GET/HEAD is performed if a  "part number" is specified in the S3
+ * query or alternatively, ranges could be specified in HTTP header
+ *
+ * @param req HTTP request
+ * @param {nb.ObjectInfo} object_md
+ * @returns an range object { start: XXX, end: YYY }
+ */
+function get_object_range(req, object_md) {
+    // S3 part number
+    if (req.query.partNumber && object_md.multipart_start !== undefined && object_md.multipart_end !== undefined) {
```

Comment:
hmm, this condition is a bit too soft - if `req.query.partNumber` is provided but `object_md.multipart_start/end` is undefined, shouldn't we error??

Reply:
Thank you for chiming in @guymguym 
The soft condition is OK, which happens in NamespaceS3, where the multipart_range is initialized from HTTP headers.

Reply:
What do you mean? Can you refer to that code?

Reply:
@guymguym Sure, [here](https://github.com/baum/noobaa-core/blob/08b3a8cd51e4b1ab65ae3ed2bf33708f80fc8460/src/sdk/namespace_s3.js#L694).